configure.ac

```diff
@@ -216,6 +216,20 @@ AC_CHECK_FILE([$srcdir/extern/fmt-5.3.0/README.rst],
                [], 
                []
                [AC_MSG_ERROR([fmt-5.3.0 is not available, please download fmt-5.3.0 from https://fmt.dev/ and put the unpacked archive in the libsemigroups/extern directory])])
+
+# From https://stackoverflow.com/q/23379805/
+
+AC_CANONICAL_HOST
+X86_CXXFLAGS=""
+AS_CASE([$host_cpu],
+        [x86_64],
```

Comment:
On BSDs, this is `amd64`. See https://en.wikipedia.org/wiki/Uname

Reply:
A better solution will just be check if the flags are supported using `AX_CHECK_COMPILE_FLAG`.